wagtail/models/__init__.py

```diff
@@ -2172,21 +2216,41 @@ class PageRevision(models.Model):
     )
 
     objects = models.Manager()
+    page_revisions = PageRevisionsManager()
     submitted_revisions = SubmittedRevisionsManager()
 
+    @cached_property
+    def content_object(self):
+        return self.base_content_type.get_object_for_this_type(pk=self.object_id)
+
+    @cached_property
+    def page(self):
+        return self.content_object
+
+    @cached_property
+    def page_id(self):
+        return int(self.object_id)
+
     def save(self, user=None, *args, **kwargs):
         # Set default value for created_at to now
         # We cannot use auto_now_add as that will override
         # any value that is set before saving
         if self.created_at is None:
             self.created_at = timezone.now()
 
+        # Set default value for base_content_type to the content_type.
+        # Page revisions should set this to the default Page model's content type,
+        # but the distinction may not be necessary for models that do not use inheritance.
+        if self.base_content_type_id is None:
+            self.base_content_type_id = self.content_type_id
+
         super().save(*args, **kwargs)
         if self.submitted_for_moderation:
-            # ensure that all other revisions of this page have the 'submitted for moderation' flag unset
-            self.page.revisions.exclude(id=self.id).update(
-                submitted_for_moderation=False
-            )
+            # ensure that all other revisions of this object have the 'submitted for moderation' flag unset
+            Revision.objects.filter(
+                content_type_id=self.content_type_id,
+                object_id=self.object_id,
+            ).exclude(id=self.id).update(submitted_for_moderation=False)
```

Comment:
Should we use `base_content_type_id` instead? Also on a few other `.filter()`s below

Reply:
The id would always be unique for a base content type, so I think so as it's technically possible that someone might change the specific content type of a page/object.